File: 567_Permutation_in_String.cpp
class Solution {
public:
    array<int, 26> count(string& str, int start, int end) {
        array<int, 26> freq = {0};
        for(int i = start; i < end; i++) 
            freq[str[i] - 'a'] ++;
        return freq;
    }
    bool checkInclusion(string s1, string s2) {
        int Len_1 = s1.length();
        int Len_2 = s2.length();
        if(Len_1 > Len_2)
            return false;
            
        array<int, 26> freq_1 = count(s1, 0, Len_1);
        array<int, 26> freq_2 = count(s2, 0, Len_1);
        for(int i = 0; i < Len_2 - Len_1; i++) {
            if(freq_1 == freq_2)
                return true;
            else {
                freq_2[s2[i] - 'a']--;
                freq_2[s2[i + Len_1] - 'a']++;
            }
        }
        if(freq_1 == freq_2)
            return true;
        return false;
    }
};

File: 567_Permutation_in_String.php
<?php
    class Solution {
    
        /**
         * @param String $s1
         * @param String $s2
         * @return Boolean
         */
        function checkInclusion($s1, $s2) {
            $Len_1 = strlen($s1);
            $Len_2 = strlen($s2);
            $index_1 = 0;
            $index_2 = 0;
            $temp_2 = $index_2;
    
            $Array_1 = ["init" => 0];
            for($i=0; $i < $Len_1; $i++){
                if(array_key_exists($s1[$i], $Array_1))
                    $Array_1[$s1[$i]] += 1;
                else
                    $Array_1[$s1[$i]] = 1;
            }
            $Copy_Array_1 = $Array_1;
    
    
            do {
                if($index_1 == $Len_1)
                    return true;
                if($Array_1[$s2[$index_2]] > 0) {
                    $Array_1[$s2[$index_2]] -= 1;
                    $index_1 ++;
                    $index_2 ++;
                }
                else {
                    $Array_1 = $Copy_Array_1;
                    $index_1 = 0;
                    $temp_2++;
                    $index_2 = $temp_2;
                }
            }while($index_2 < $Len_2);
    
            if($index_1 == $Len_1)
                return true;
            return false;
        }
    }
?>
